### ISSUE TITLE
switch_to_blog() - flag for missing subsequent restore_current_blog()

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#651]
## Describe the solution you'd like

From https://github.com/Automattic/VIP-Coding-Standards/pull/649#discussion_r606174833, it would be worth flagging that `restore_current_blog()` should be called subsequently (if not already). 

## What code should be reported as a violation?

```
function switch_blog() {
	$id = get_current_blog_id();

	if ( $id === 2 ) {
		return;
	}

	switch_to_blog( 2 );
}
```

## What code should *not* be reported as a violation?

```
function switch_blog() {
	// do stuff.
	$id = get_current_blog_id();

	if ( $id === 2 ) {
		return;
	}

	switch_to_blog( 2 );
	restore_current_blog();
}
```

**Comment by jrfnl** [comment]
Note: while I think it would be good to add a check for this, it should be a `warning` and the checking for `restore_current_blog()` should be limited to the scope in which the call to `switch_to_blog()` is found.

Any such sniff will always have the potential for false positives as a plugin can have a wrapper `PluginClass::restore_stuff()` function declared in another file and we won't be able to trace the function calls through the code base/across files.

All the same, it is _best practice_ to always call `restore_current_blog()` after a `switch_to_blog()` _within the same function_, so the fact that the sniff has the risk of false positives should not be a deterrent for creating it as the potential for bugs when the call is not made far outweighs the annoyance of false positives from a sniff.

**Comment by rebeccahum** [comment]
https://github.com/Automattic/VIP-Coding-Standards/pull/649#discussion_r613558501 You mentioned that you can see this being a good sniff for WPCS — is this something you'd like me to open an issue there for, @jrfnl?

**Comment by jrfnl** [comment]
@rebeccahum I wouldn't be surprised if there is an issue open for it already, but yeah, go ahead, though search in the repo first for existing issues.
